testkit/gradle-testkit-plugin/src/main/kotlin/com/autonomousapps/GradleTestKitPlugin.kt

```diff
@@ -53,8 +53,14 @@ public class GradleTestKitPlugin : Plugin<Project> {
           // Gradle tests generally require more metaspace
           jvmArgs("-XX:+HeapDumpOnOutOfMemoryError", "-XX:MaxMetaspaceSize=1g")
 
-          systemProperty("com.autonomousapps.plugin-under-test.repo", extension.funcTestRepo.absolutePath)
-          systemProperty("com.autonomousapps.plugin-under-test.version", version.toString())
+          systemProperty(
+            "com.autonomousapps.plugin-under-test.repo",
+            target.provider { extension.funcTestRepo.absolutePath }.get()
+          )
+          systemProperty(
+            "com.autonomousapps.plugin-under-test.version",
+            target.provider { version.toString() }.get()
+          )
```

Comment:
I found that setting the system properties was too eager in a project I'm working on. The version hasn't been set yet when the plugin is applied and the extension hasn't been configured by the time this task is registered.